File: backend/app/schemas/achievement.py
import uuid
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict


class AchievementOut(BaseModel):
    id: uuid.UUID
    code: str
    title: str
    description: str
    category: str
    icon_name: str
    condition_type: str
    condition_threshold: int
    reward_xp: int
    reward_gold: int
    reward_title: Optional[str] = None
    is_unlocked: bool = False
    unlocked_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/achievement.ts
export interface Achievement {
  id: string;
  code: string;
  title: string;
  description: string;
  category: string;
  icon_name: string;
  condition_type: string;
  condition_threshold: number;
  reward_xp: number;
  reward_gold: number;
  reward_title: string | null;
  is_unlocked: boolean;
  unlocked_at: string | null;
}
